opensearch_py_ml/sentence_transformer_model/sentencetransformermodel.py

```diff
@@ -0,0 +1,677 @@
+        Description:
+        Create input data for training
+
+        Parameters:
+        df: pd.DataFrame
+            required for loading sentence transformer examples.
+        use_accelerate: bool = False,
+            Optional, use accelerate to fine tune model. Default as false to not use accelerator.
+        Return:
+            the list of train examples.
+        """
+
+        train_examples = []
+        print("Loading training examples... ")
+
+        if use_accelerate is False:
+            for i in tqdm(range(len(df)), total=len(df)):
+                train_examples.append(
+                    InputExample(
+                        texts=[
+                            df[i : i + 1]["passages"].values[0],
+                            df[i : i + 1]["query"].values[0],
+                        ]
+                    )
+                )
+        else:
+            #  ML: broken down the class into following
+            #  train_examples = tasb_dataset_vanilla(df)
+            queries = list(df["query"])
+            passages = list(df["passages"])
+            for i in tqdm(range(len(df)), total=len(df)):
+                train_examples.append([queries[i], passages[i]])
+        return train_examples
+
+    def train_model(
+        self,
+        train_examples: List[str],
+        model_url: str = None,
+        output_path: str = None,
+        output_model_name: str = None,
+        use_accelerate: bool = False,
+        learning_rate: float = 2e-5,
+    ):
+        """
+        Description:
+        Takes in training data and a sentence transformer url to train a custom semantic search model
+
+        Parameters:
+        train_examples:
+            required, input for the sentence transformer model training
+        model_url: str = None
+            optional,the url to download sentence transformer model, if None, default as 'sentence-transformers/msmarco-distilbert-base-tas-b'
+        output_path: str=None
+            optional,the path to store trained custom model. If None, default as current folder path
+        output_model_name: str=None
+            optional,the name of the trained custom model. If None, default as 'trained_model.pt'
+        use_accelerate: bool = False,
+            Optional, use accelerate to fine tune model. Default as false to not use accelerator.
+        learning_rate: float
+            optional, learning rate to train model, default is 2e-5
+        Return:
+            None
+        """
+
+        if output_path is None:
+            output_path = os.getcwd()
+        if output_model_name is None:
+            output_model_name = "trained_model.pt"
+        if model_url is None:
+            model_url = "sentence-transformers/msmarco-distilbert-base-tas-b"
+
+        # prepare an output model path for later saving the pt model.
+        output_model_path = output_path + "/" + output_model_name
+
+        # declare variables before assignment for training
+        num_epochs = 20
```

Comment:
I think we should add this in the parameter to give customer flexibility to add more or less epoch if they want. 

We can add default value 20 in the parameter.

Reply:
discussed with Milind about training parameters, we decided NOT to open the parameters for training purpose, the training function can be modified in local folder if data scientist would like to fine tune in a customized way.

Reply:
added num_epochs  in 
[added save_as_pt function](https://github.com/opensearch-project/opensearch-py-ml/pull/18/commits/094339cca0a561d3ecd4edf74ce803b77e973239)